client/my-sites/plans/current-plan/jetpack-product-install/index.js

```diff
@@ -0,0 +1,239 @@
+
+/**
+ * Module variables
+ */
+/**
+ * These are plugin states in the installation lifecycle we consider "non-error" states.
+ */
+const NON_ERROR_STATES = [
+	'not_active', // Plugin is not installed
+	'option_name_not_in_whitelist', // Plugin is installed but not activated
+	'key_not_set', // Plugin is installed and activated, but not configured
+	'installed', // Plugin is installed, activated and configured
+];
+/**
+ * Those errors are any of the following:
+ * - Temporary, occurring if we request installation status while plugin is being set up.
+ * - Permanent, occurring if there is a failure we can't fix by waiting.
+ * We attempt to recover from these errors by retrying status requests.
+ */
+const RECOVERABLE_ERROR_STATES = [ 'vaultpress_error' ];
+/**
+ * The plugins this product installer installs, activates and configures.
+ */
+const PLUGINS = [ 'akismet', 'vaultpress' ];
+/**
+ * Maximum number of attempts to refetch installation status in the event of a recoverable error.
+ */
+const MAX_RETRIES = 3;
+
+export class JetpackProductInstall extends Component {
+	state = {
+		startedInstallation: false,
+	};
+
+	retries = 0;
+
+	componentDidMount() {
+		this.requestInstallationStatus();
+		this.maybeStartInstall();
+	}
+
+	componentDidUpdate() {
+		this.maybeStartInstall();
+	}
+
+	/**
+	 * Start the plugin installation if all conditions are matched:
+	 * - Installation hasn't started yet.
+	 * - We don't have any non-recoverable installation errors.
+	 * - We have already fetched the plugin installation status.
+	 * - Installation has not finished.
+	 * - We already have the plugin keys.
+	 */
+	maybeStartInstall() {
+		const { pluginKeys, progressComplete, siteId } = this.props;
+
+		// We're already installing
+		if ( this.state.startedInstallation ) {
+			return;
+		}
+
+		// We have non-recoverable installation errors
+		if ( this.installationHasErrors() && ! this.installationHasRecoverableErrors() ) {
+			return;
+		}
+
+		// We need to wait for installation status first
+		if ( progressComplete === null ) {
+			return;
+		}
+
+		// We should skip install if everything is already installed and configured
+		if ( progressComplete === 100 ) {
+			return;
+		}
+
+		// Installation can be started only if we have the Akismet and VaultPress keys
+		if ( pluginKeys && pluginKeys.akismet && pluginKeys.vaultpress ) {
+			this.setState( {
+				startedInstallation: true,
+			} );
+
+			this.props.startJetpackProductInstall( siteId, pluginKeys.akismet, pluginKeys.vaultpress );
+		}
+	}
+
+	/**
+	 * Used to determine if at least one plugin is in at least one of the provided plugin states.
+	 *
+	 * @param {Array} pluginStates States to check against.
+	 * @return {Boolean} True if at least one plugin is in at least one of the given states, false otherwise.
+	 */
+	arePluginsInState( pluginStates ) {
+		const { status } = this.props;
+
+		if ( ! status ) {
+			return false;
+		}
+
+		return some( PLUGINS, pluginSlug => {
```

Comment:
No particular reason, just ended up in this version because I was concentrating on bigger things :) 

But yeah, @ockham caught that too, fixed in 3b27ec2b9ece63364585fa99132245149b5a0cf0